### PR TITLE
add (failing) tests for unauthorized ec2 crud

### PR DESCRIPTION
**Pull request by anotherjesse**
Currently the test fails with:

AuthorizationFailure: Authorization Failed: Unable to communicate with identity service: Traceback (most recent call last)

It seems to be making the request with:

```
root: DEBUG: {"auth": {"passwordCredentials": {"username": "BOO", "password": "boo2"}, "tenantId": "baz"}}
```

But keystone is blowing up with:

```
File "/Users/jesse/rcb/keystonelight/keystone/service.py", line 587, in authenticate
  for role_id in metadata_ref.get('roles', []):
AttributeError: 'NoneType' object has no attribute 'get'
```

### CODE REVIEW

**Comment by anotherjesse** [comment]
The original issue appears to be because:

```
diff --git a/tests/default_fixtures.py b/tests/default_fixtures.py
index 82a0d96..990e241 100644
--- a/tests/default_fixtures.py
+++ b/tests/default_fixtures.py
@@ -10,6 +10,7 @@ USERS = [

 METADATA = [
     {'user_id': 'foo', 'tenant_id': 'bar', 'extra': 'extra'},
+    {'user_id': 'boo', 'tenant_id': 'baz', 'extra': 'extra'},
     ]

 ROLES = [
```

The test still doesn't pass but now you can see something odd (all the "extra" stuff in the response)

```
keystoneclient.client: DEBUG: REQ: curl -i http://localhost:53277/v2.0/tokens -X POST -H "Content-Type: application/json" -H "User-Agent: python-keystoneclient"

keystoneclient.client: DEBUG: REQ BODY: {"auth": {"passwordCredentials": {"username": "BOO", "password": "boo2"}, "tenantId": "baz"}}

keystoneclient.client: DEBUG: RESP: {'date': 'Wed, 18 Jan 2012 05:22:09 GMT', 'status': '200', 'content-length': '999', 'content-type': 'text/html; charset=UTF-8'}
RESP BODY: {"access": {"token": {"expires": "", "id": "7e91a1b6ffdf4ffdb24772610f39d608", "tenant": {"enabled": true, "id": "baz", "name": "BAZ", "extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"enabled": true, "extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {"extra": {}}}}}}}}}}}}}}}}}}}}}}}, "serviceCatalog": [{"endpoints": [{"adminURL": "http://localhost:3000/v1.1/baz", "region": "RegionOne", "publicURL": "http://localhost:3000/v1.1/baz", "internalURL": "http://localhost:3000/v1.1/baz"}], "endpoints_links": [], "type": "compute", "name": "'Compute Service'"}, {"endpoints": [{"adminURL": "http://localhost:53278/v2.0", "region": "RegionOne", "publicURL": "http://localhost:53277/v2.0", "internalURL": "http://localhost:53278/v2.0"}], "endpoints_links": [], "type": "identity", "name": "'Identity Service'"}], "user": {"username": "BOO", "roles_links": [], "id": "boo", "roles": [], "name": "BOO"}}}
```

File: tests/test_keystoneclient.py [hunk 1]
```diff
@@ -65,13 +65,17 @@ def _config(self):
         CONF(config_files=[test.etcdir('keystone.conf'),
                            test.testsdir('test_overrides.conf')])
 
-    def foo_client(self):
-        return self._client(username='FOO',
-                            password='foo2',
-                            tenant_name='BAR')
+    def get_client(self, username='FOO'):
```
Comment (termie) [inline tests/test_keystoneclient.py]: i'd prefer to send an actual user, so you'd call self.get_client(self.user_foo)

probably done as

def get_client(self, user_ref, tenant_ref):

Comment (anotherjesse) [inline tests/test_keystoneclient.py]: done - it isn't pretty but we can keep cleaning it up as we go?

**Comment by devcamcar** [comment]
lgtm mate

File: tests/backend_sql.conf [hunk 2]
```diff
@@ -7,3 +7,6 @@ pool_timeout = 200
 
 [identity]
 driver = keystone.identity.backends.sql.Identity
+
+[ec2]
+driver = keystone.contrib.ec2.backends.SqlEc2
```
Comment (termie) [inline tests/backend_sql.conf]: SqlEc2 -> sql.Ec2

File: tests/test_keystoneclient.py [hunk 3]
```diff
@@ -65,13 +65,22 @@ def _config(self):
...
 
-    def foo_client(self):
-        return self._client(username='FOO',
-                            password='foo2',
-                            tenant_name='BAR')
+    def get_client(self, user_ref=None, tenant_ref=None):
+        if user_ref is None:
+            user_ref = self.user_foo
+        if tenant_ref is None:
+            for user in default_fixtures.USERS:
```
Comment (termie) [inline tests/test_keystoneclient.py:72]: i don't think you should be referencing default_fixtures in here, but i'm willing to leave it for now as this is just a helper function

**Comment by termie** [comment]
looks good, merging

File: tests/backend_sql.conf (hunk 2 above)
Comment (anotherjesse) [inline tests/backend_sql.conf]: switched to:

```
driver = keystone.contrib.ec2.backends.sql.Ec2
```

but it still fails to run the sql tests - I did something bad...

**Pull request merged**